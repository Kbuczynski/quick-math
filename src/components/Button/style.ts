import { Text, TouchableOpacity } from 'react-native'
import styled from 'styled-components'

export const StyledBtn = styled(TouchableOpacity)<{ isPartOfList: boolean }>`
  padding: 12px 8px;
  border: 1px solid #3b8e94;
  background-color: #000;
  justify-content: center;
  align-items: center;
  ${(props) => props.isPartOfList && 'flex: 1;'}
`

export const StyledBtnLabel = styled(Text)<{ disabled: boolean }>`
  font-size: 28px;
  color: #fff;
  ${(props) => props.disabled && 'opacity: 0.5;'}
`
