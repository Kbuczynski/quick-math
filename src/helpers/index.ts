export * from './getRandomNumber'
export * from './isDivideBy'
