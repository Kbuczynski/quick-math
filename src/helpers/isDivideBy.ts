export function isDivideBy(a: number, b: number): boolean {
  return a % b === 0
}
