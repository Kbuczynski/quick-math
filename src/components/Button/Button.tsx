import { StyledBtn, StyledBtnLabel } from './style'

interface ButtonProps {
  label: string
  handlePress: () => void
  disabled?: boolean
  isPartOfList?: boolean
  testId?: string
}

export default function Button({ label, handlePress, disabled = false, isPartOfList = false, testId }: ButtonProps) {
  return (
    <StyledBtn
      onPress={disabled ? () => null : handlePress}
      disabled={disabled}
      isPartOfList={isPartOfList}
      testID={testId}
    >
      <StyledBtnLabel disabled={disabled}>{label}</StyledBtnLabel>
    </StyledBtn>
  )
}
