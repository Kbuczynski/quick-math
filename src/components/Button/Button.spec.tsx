import Button from './Button'
import { renderWithTheme } from '../../helpers/tests'
import { fireEvent } from '@testing-library/react-native'
import { ReactTestInstance } from 'react-test-renderer'

const buttonTestId = 'button'
const handlePressMock = jest.fn()
const labelMock = 'test'

describe('Button:', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should render and not be disabled', async () => {
    const screen = renderWithTheme(<Button label={labelMock} handlePress={handlePressMock} testId={buttonTestId} />)
    const button = screen.getByTestId(buttonTestId)
    const buttonLabel = button.children[0] as ReactTestInstance

    fireEvent.press(button)
    fireEvent.press(buttonLabel)
    expect(handlePressMock).toBeCalledTimes(2)
    expect(screen.getByText(labelMock)).toBeTruthy()
    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('should render and be disabled', async () => {
    const screen = renderWithTheme(
      <Button label={labelMock} handlePress={handlePressMock} disabled testId={buttonTestId} />,
    )
    const button = screen.getByTestId(buttonTestId)
    const buttonLabel = button.children[0] as ReactTestInstance

    fireEvent.press(button)
    fireEvent.press(buttonLabel)
    expect(handlePressMock).not.toBeCalled()
    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('should render and be a part of the list', () => {
    const screen = renderWithTheme(
      <Button label={labelMock} handlePress={handlePressMock} isPartOfList testId={buttonTestId} />,
    )
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
