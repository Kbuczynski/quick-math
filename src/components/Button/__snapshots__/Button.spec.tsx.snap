// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button: should render and be a part of the list 1`] = `
<View
  accessibilityState={
    {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#141414",
      "borderColor": "#3B8E94",
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 12,
    }
  }
  testID="button"
>
  <Text
    disabled={false}
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 28,
        },
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`Button: should render and be disabled 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#141414",
      "borderColor": "#3B8E94",
      "borderStyle": "solid",
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 12,
    }
  }
  testID="button"
>
  <Text
    disabled={true}
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 28,
          "opacity": 0.5,
        },
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`Button: should render and not be disabled 1`] = `
<View
  accessibilityState={
    {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#141414",
      "borderColor": "#3B8E94",
      "borderStyle": "solid",
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 12,
    }
  }
  testID="button"
>
  <Text
    disabled={false}
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 28,
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
